bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -0,0 +1,371 @@
+        Files.createDirectories(deploymentDir);
+        // Exploded deployment
+        boolean isExploded = Files.isDirectory(deployment);
+        updateConfig(jbossHome.resolve(STANDALONE).resolve(CONFIGURATION).resolve(STANDALONE_CONFIG),
+                deployment.getFileName().toString(), isExploded);
+        if (isExploded) {
+            copyDirectory(deployment, target);
+        } else {
+            Files.copy(deployment, target);
+        }
+        log.installDeployment(deployment);
+    }
+
+    private static void updateConfig(Path configFile, String name, boolean isExploded) throws Exception {
+        FileInputStream fileInputStream = new FileInputStream(configFile.toFile());
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+
+        Document document = documentBuilder.parse(fileInputStream);
+        Element root = document.getDocumentElement();
+
+        NodeList lst = root.getChildNodes();
+        for (int i = 0; i < lst.getLength(); i++) {
+            Node n = lst.item(i);
+            if (n instanceof Element) {
+                if (DEPLOYMENTS.equals(n.getNodeName())) {
+                    throw BootableJarLogger.ROOT_LOGGER.deploymentAlreadyExist();
+                }
+            }
+        }
+        Element deployments = document.createElement(DEPLOYMENTS);
+        Element deployment = document.createElement(DEPLOYMENT);
+        Element content = document.createElement(CONTENT);
+        content.setAttribute(SHA1, DEP_1 + DEP_2);
+        if (isExploded) {
+            content.setAttribute(ARCHIVE, "false");
+        }
+        deployment.appendChild(content);
+        deployment.setAttribute(NAME, name);
+        deployment.setAttribute(RUNTIME_NAME, name);
+        deployments.appendChild(deployment);
+
+        root.appendChild(deployments);
+        Transformer transformer = TransformerFactory.newInstance().newTransformer();
+        StreamResult output = new StreamResult(configFile.toFile());
+        DOMSource input = new DOMSource(document);
+
+        transformer.transform(input, output);
+
+    }
+
+    private void copyDirectory(Path src, Path target) throws IOException {
+        Files.walk(src).forEach(file -> {
+            try {
+                Path targetFile = target.resolve(src.relativize(file));
+                if (Files.isDirectory(file)) {
+                    if (!Files.exists(targetFile)) {
+                        Files.createDirectory(targetFile);
+                    }
+                } else {
+                    Files.copy(file, targetFile);
+                }
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+        });
+    }
+
+    private void configureLogger() throws IOException {
+        System.setProperty(LOG_MANAGER_PROP, LOG_MANAGER_CLASS);
```

Comment:
If we do this we mandate that users must use the JBoss Log Manager. Maybe this is okay, but maybe it's something we should look at.

Reply:
@jamezp , I was under the assumption that the jboss logmanager was the only usable implementation with the Logging subsystem. You are suggesting that user could set the system property and not rely on the logging subsystem? I could check if the system property exists and do nothing if that is the case.

Reply:
I'm redacting this concern. The logging subsystem does require the JBoss Log Manager, my thought was if the subsystem was removed maybe the user would want to use a different log manager. However the servers main entry point does do some log manager initialization as well so removing a requirement like that is out of scope for this. Sorry about the confusion there.